### PR TITLE
When invoking an exception view, push the new threadlocals

### PR DESCRIPTION
**Pull request by bertjwregeer**
This way when calling the threadlocal get_current_request() you get the
same request object as the one that was passed to the view.

Fixes the comment on: https://github.com/Pylons/pyramid/issues/3053#issuecomment-303520185

### CODE REVIEW

File: pyramid/view.py [hunk 1]
```diff
@@ -690,17 +699,23 @@ def invoke_exception_view(
...
-                context_iface,
-                '',
-                view_types=None,
-                view_classifier=IExceptionViewClassifier,
-                secure=secure,
-                request_iface=request_iface.combined,
-                )
+
+            try:
+                manager.push({'request': request, 'registry': registry})
```
Comment (mmerickel) [inline pyramid/view.py]: This `push` should go before the `try`. You do not want to try to `pop` if the `push` fails. It's pedantic but could avoid raising an exception from an exception.

Comment (bertjwregeer) [inline pyramid/view.py]: Gotcha. Will make the change.

Comment (mmerickel) [inline pyramid/view.py]: *reminder bump*

Comment (bertjwregeer) [inline pyramid/view.py]: Done.

**Pull request merged**